sdk/kubeflow/trainer/types/types.py

```diff
@@ -50,39 +50,29 @@ class TrainJob:
     status: Optional[str] = "Unknown"
 
 
-# Configuration for the Lora to configure parameter efficient fine-tuning.
+# Configuration for the custom trainer.
 @dataclass
-class LoraConfig:
-    r: Optional[int] = field(
-        default=None, metadata={"help": "Lora attention dimension"}
-    )
-    lora_alpha: Optional[int] = field(default=None, metadata={"help": "Lora alpha"})
-    lora_dropout: Optional[float] = field(
-        default=None, metadata={"help": "Lora dropout"}
-    )
-
-
-@dataclass
-class FineTuningConfig:
-    # TODO (andreyvelich): Add more configs once we support them, e.g. QLoRA.
-    peft_config: Optional[LoraConfig] = None
-
-
-# Configuration for the Trainer.
-# TODO (andreyvelich): Discuss what values should be on the Trainer.
-@dataclass
-class Trainer:
-    """Trainer configuration.
-    TODO: Add the description
+class CustomTrainer:
+    """Custom Trainer configuration. Configure the self-contained function
+        that encapsulates the entire model training process.
+
+    Args:
+        func (`Optional[Callable]`):
+            The function that encapsulates the entire model training process.
+        func_args (`Optional[Dict]`): The arguments to pass to the function.
+        packages_to_install (`Optional[List[str]]`):
+            A list of Python packages to install before running the function.
+        pip_index_url (`str`): The PyPI URL from which to install Python packages.
+        num_nodes (`Optional[int]`): The number of nodes to use for training.
+        resources_per_node (`Optional[Dict]`): The computing resources to allocate per node.
     """
 
     func: Optional[Callable] = None
     func_args: Optional[Dict] = None
     packages_to_install: Optional[List[str]] = None
     pip_index_url: str = constants.DEFAULT_PIP_INDEX_URL
```

Comment:
Should we make this Optional as well ?
```suggestion
    pip_index_url: Optional[str] = constants.DEFAULT_PIP_INDEX_URL
```

Reply:
Yeah, I think so. Or users may get confused about the type of `pip_index_url` in the description.

Reply:
@Electronic-Waste Did you get a chance to push this change, so we can merge your PR ?

Reply:
I'll push it soon. There is some network error on my server:)

Reply:
Sure, thank you for your time!

Reply:
@andreyvelich I've pushed this change. Thanks for your detailed review!